thentos-core/src/Thentos/Frontend/State.hs

```diff
@@ -137,21 +127,21 @@ enterFAction aState key smap = Nat $ ExceptT . (>>= _Left fActionServantErr) . r
     run fServer = fst <$> runActionE emptyFrontendSessionData aState fServer'
       where
         fServer' :: FAction a
-        fServer' = do
+        fServer' =
             case smap key of
-                Nothing -> error $ "enterFAction: internal error in servant-session: no cookie!"
+                Nothing ->
+                    -- Should really this be 'error' here.
+                    -- Can a misbehaving client trigger this error? Thus crashing the server?
```

Comment:
nope, changed my mind: you are right and this should be a more graceful failure.  somehting like `redirect /no-cookie`.

I'll add a FIXME.